src/js/pwm.js

```diff
@@ -66,29 +66,21 @@ function PWM(pin, options, callback) {
   // Check arguments.
   if (util.isFunction(options)) {
     callback = options;
-    self._options.dutyCycle = -1;
-    self._options.period = -1;
   } else if (util.isObject(options)) {
-    if (util.isNumber(options.dutyCycle)) {
-      self._options.dutyCycle = options.dutyCycle;
-    } else {
-      self._options.dutyCycle = -1;
+    var dutyCycle = options.dutyCycle;
+    var period = options.period;
+    if (!util.isNumber(period) && util.isNumber(options.frequency)) {
+      period = 1.0 / options.frequency;
     }
 
-    if (util.isNumber(options.period)) {
+    if (util.isNumber(dutyCycle) && dutyCycle >= 0.0 && dutyCycle <= 1.0 &&
```

Comment:
No. It is okay to grab the PWM instance and not set it up initially, so no exception throwing at this point.

Only when the period AND duty cycle is set, a request to the hardware driver is made.